traffic_ops/testing/compare/compare.go

```diff
@@ -157,74 +159,301 @@ func testRoute(tos []*Connect, route string) {
+		log.Fatalf("Failed to parse response body from %s/%s as JSON: %s\n", (*responses)[1].TO.URL, route, err.Error())
+	}
+
+	// type-safety cast to maps
+	m0, m1 := result0.(map[string]interface{}), result1.(map[string]interface{})
+
+	// ... Now we have to iterate over every key in each map to determine if it should be removed
+	for key, value := range m0 {
+
+		// handles snapshot/new timestamps/hostnames/versions
+		if (key == "response") {
+			switch Type := value.(type) {
+				case map[string]interface{}:
+					if k, in := value.(map[string]interface{})["stats"]; in {
+						switch t := k.(type) {
+							case map[string]interface{}:
+								if val, ok := k.(map[string]interface{})["date"]; ok {
+									switch T := val.(type) {
+										case float64:
+											delete(result0.(map[string]interface{})["response"].(map[string]interface{})["stats"].(map[string]interface{}), "date")
+										default:
+											log.Printf("found 'stats.date' object - wasn't an integer - %v", T)
+									}
+								}
+
+								if val, ok := k.(map[string]interface{})["tm_host"]; ok {
+									switch T := val.(type) {
+										case string:
+											delete(result0.(map[string]interface{})["response"].(map[string]interface{})["stats"].(map[string]interface{}), "tm_host")
+										default:
+											log.Printf("found 'stats.tm_host' object - wasn't a string - %v", T)
+									}
+								}
+
+								if val, ok := k.(map[string]interface{})["tm_version"]; ok {
+									switch T := val.(type) {
+										case string:
+											delete(result0.(map[string]interface{})["response"].(map[string]interface{})["stats"].(map[string]interface{}), "tm_version")
+										default:
+											log.Printf("found 'stats.tm_version' object - wasn't a string - %v", T)
+									}
+								}
+							default:
+								log.Printf("found 'stats' object - wasn't a map - %v\n", t)
+						}
+					}
+				default:
+					log.Printf("found 'response' object - wasn't a map - %v\n", Type)
+			}
+
+		// Handles hostname differences in api/1.x/servers/{{server}}/configfiles/ats endpoints
+		} else if (key == "info") {
+			switch t := value.(type) {
+				case map[string]interface{}:
+					if val, ok := value.(map[string]interface{})["toUrl"]; ok {
+						switch T := val.(type) {
+							case string:
+								delete(result0.(map[string]interface{})["info"].(map[string]interface{}), "toUrl")
+							default:
+								log.Printf("found 'info.toUrl' object - wasn't a string - %v", T)
+						}
+					}
+
+					if val, ok := value.(map[string]interface{})["toRevProxyUrl"]; ok {
+						switch T := val.(type) {
+							case string:
+								delete(result0.(map[string]interface{})["info"].(map[string]interface{}), "toRevProxyUrl")
+							default:
+								log.Printf("found 'info.toRevProxyUrl' object - wasn't a string - %v", T)
+						}
+					}
+				default:
+					log.Printf("found 'info' object - wasn't a map - %v", t)
+			}
+		}
+
```

Comment:
yeah -- definitely indentation issue here..    go fmt

Reply:
Go doesn't think `case`s should be indented...